### PR TITLE
Issue #353: Merges JSON and Exec Params files

### PR DESCRIPTION
**Pull request by robertojrojas**
Issue #353: Merges JSON and Exec Params files

### CODE REVIEW

File: cfg/1.11/node.yaml [hunk 1]
```diff
@@ -26,17 +26,19 @@ groups:
       systemctl daemon-reload
       systemctl restart kubelet.service
     scored: true
-
   - id: 2.1.2
```
Comment (lizrice) [inline cfg/1.11/node.yaml]: Not a giant deal but I think it's a shame for readability to lose the blank lines like the one that was at line 29

File: check/check.go [hunk 2]
```diff
@@ -60,21 +63,29 @@ func handleError(err error, context string) (errmsg string) {
...
 	Scored         bool   `json:"scored"`
 	ExpectedResult string `json:"expected_result"`
 }
 
+type AuditCommands struct {
+	FromConfig string      `yaml:"from_config"`
+	FromParams string      `yaml:"from_params"`
+	ConfigCmds []*exec.Cmd `json:"omit"`
+	ParamsCmds []*exec.Cmd `json:"omit"`
+}
```
Comment (lizrice) [inline check/check.go]: I think it would be neater to just add a new pair of fields into the original `Check` struct so that it's more like 

```
...
Audit       string `json:"audit"`
AuditConfig string `json:"audit_config"`
...
Commands []*exec.Cmd `json:"omit"`
ConfigCmds []*exec.Cmd `json:"omit"`
...
```

That way the back-compatibility comes essentially for free - if there is no `audit_config` field in the test definition you can simply use exactly the code as-is

File: cfg/1.11/config.yaml [hunk 3]
```diff
@@ -1,2 +1,15 @@
 ---
 ## Version-specific settings that override the values in cfg/config.yaml
+node:
```
Comment (lizrice) [inline cfg/1.11/config.yaml]: Are these really specific to 1.11, or should they be moved to `cfg/config.yaml`. Iirc these values were originally added to match the settings on EKS.

File: cfg/1.11/node.yaml [hunk 4]
```diff
@@ -1,23 +1,23 @@
 ---
 controls:
-version: 1.11
-id: 2
-text: "Worker Node Security Configuration"
+id: "2"
+version: "1.11"
```
Comment (lizrice) [inline cfg/1.11/node.yaml]: Doesn't matter much, but I think it reads better to have the version field ahead of the id

File: check/check_test.go [hunk 5]
```diff
@@ -25,11 +25,11 @@ func TestCheck_Run(t *testing.T) {
 	}
 
 	testCases := []TestCase{
```
[the remark this answers is not in the document]
Comment (lizrice) [reply check/check_test.go]: These tests all pass, but if you run with debug logging you can see that we attempt to run an audit command even if it doesn't exist: 

```
$ go test -v -run TestCheckAuditConfig ./check -args -v 3 -logtostderr true
=== RUN   TestCheckAuditConfig
I1003 10:52:05.313596   95098 check.go:182] Check.ID: 0 Command: "echo flag=correct" TestResult: true Score: "PASS" 
I1003 10:52:05.343462   95098 check.go:182] Check.ID: 3 Command: "echo 'flag: correct'" TestResult: true Score: "PASS" 
I1003 10:52:05.364571   95098 check.go:182] Check.ID: 5 Command: "echo flag=correct" TestResult: true Score: "PASS" 
I1003 10:52:05.388908   95098 check.go:182] Check.ID: 7 Command: "echo 'flag: correct'" TestResult: true Score: "PASS" 
I1003 10:52:05.388928   95098 check.go:196] failed to run: 
failed command: [], error: fork/exec : no such file or directory
failed to run: 
failed command:[], error: exec: not started
I1003 10:52:05.402864   95098 check.go:196] failed to run: 
failed command: [], error: fork/exec : no such file or directory
failed to run: 
failed command:[], error: exec: not started
--- PASS: TestCheckAuditConfig (0.10s)
PASS
ok      github.com/aquasecurity/kube-bench/check        0.116s
```

I think we shouldn't be trying to run audit or audit_config if they are empty, wdyt?

[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check_test.go]: Thanks for adding the tests! You're right it should be trying to run the checks if both audit and audit_config are empty. I'll take a look at.

File: check/check_test.go [hunk 6]
```diff
@@ -25,11 +25,11 @@ func TestCheck_Run(t *testing.T) {
...
-		{check: Check{Type: "manual"}, Expected: WARN},
+		{check: Check{Type: MANUAL}, Expected: WARN},
 		{check: Check{Type: "skip"}, Expected: INFO},
 		{check: Check{Type: "", Scored: false}, Expected: WARN}, // Not scored checks with no type should be marked warn
 		{check: Check{Type: "", Scored: true}, Expected: WARN},  // If there are no tests in the check, warn
-		{check: Check{Type: "manual", Scored: false}, Expected: WARN},
+		{check: Check{Type: MANUAL, Scored: false}, Expected: WARN},
 		{check: Check{Type: "skip", Scored: false}, Expected: INFO},
 	}
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check_test.go]: Thanks for these tests!

File: check/check.go [hunk 7]
```diff
@@ -119,116 +113,82 @@ func (c *Check) run() State {
...
+	state, finalOutput, retErrmsgs := performAuditTest(c.Audit, c.Commands, currentTests)
+	if len(state) > 0 {
+		c.State = state
+		return c.State
+	}
+	errmsgs := retErrmsgs
+
+	// If something went wrong with the 'Audit' command
+	// and an 'AuditConfig' command was provided, use it to
+	// perform the Check/Test
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check.go]: I'm thinking in refactoring all this for the bench-common integration as this changes will need to be incorporated there.

File: check/check.go [hunk 8]
```diff
@@ -103,109 +108,88 @@ func (c *Check) run() State {
...
-		)
-
-		if i < n-1 {
-			cs[i].Stdout.(io.Closer).Close()
+	errmsgs := retErrmsgs
+
+	// If something went wrong with the 'Audit' command
+	// and an 'AuditConfig' command was provided, use it to
+	// perform the Check/Test
+	if (finalOutput == nil || !finalOutput.testResult) && hasAuditConfig {
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check.go]: Yes, that's what I was thinking as well.

File: check/check.go [hunk 9]
```diff
@@ -300,6 +260,27 @@ func isShellCommand(s string) bool {
 	return false
 }
 
+func performAuditTest(audit string, commands []*exec.Cmd, tests *tests) (State, *testOutput, string) {
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check.go]: Yes, I agree. That's better.

File: check/check.go [hunk 10]
```diff
@@ -331,46 +310,51 @@ func runExecCommands(audit string, commands []*exec.Cmd, out *bytes.Buffer) (Sta
...
-		)
+		if err != nil {
+			errmsgs += handleError(
+				err,
+				fmt.Sprintf("failed to run: %s\nfailed command: %s",
+					audit,
+					cs[i].Args,
+				),
+			)
+		}
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply check/check.go]: You're right. Removing handleError.

File: check/check.go [hunk 11]
```diff
@@ -274,3 +244,86 @@ func isShellCommand(s string) bool {
...
+			errmsgs += fmt.Sprintf("failed to run: %s, command: %s, error: %s\n", audit, cs[i].Args, err)
+		}
+
+		if i < n-1 {
+			cs[i].Stdout.(io.Closer).Close()
+		}
+		i++
+	}
+
+	glog.V(9).Infof("Command %q - Output:\n\n %s\n", audit, out.String())
```
Comment (lizrice) [inline check/check.go]:
```suggestion
	glog.V(3).Infof("Command %q - Output:\n\n %s\n", audit, out.String())
```

Comment (lizrice) [reply check/check.go]: This is roughly the equivalent of what was logged at line 187 in the original, right?
